File: php/run-test.php
<?php
error_reporting(0);
require_once __DIR__ . "/vendor/autoload.php";
require_once __DIR__ . "/lib/common_wrappers.php";
use UnificationEngine\Models\UEApp;
use UnificationEngine\Models\UEUser;

$conf = json_decode(file_get_contents(__DIR__ . "/../common/conf.json"), true);

$app = new UEApp($conf["APP_KEY"],$conf["APP_SECRET"]);

// CREATE USER
try {
  ob_start();
  $user = $app->create_user();
  ob_end_clean();
  print_success("Create User");
}
catch(Exception $e){
  ob_end_clean();
  print_error("Create User");
  echo "\t" . $e->getMessage() . "\n";
}


// ADD FACEBOOK CONNECTION
try {
  ob_start();
  $fb_connection = $user->add_connection("fb","facebook",$conf["SERVICE_TOKENS"]["FACEBOOK"]);
  ob_end_clean();
  print_success("Add Facebook Connection");
}
catch(Exception $e){
  ob_end_clean();
  print_error("Add Facebook Connection");
  echo "\t" . $e->getMessage() . "\n";
}


// ADD TWITTER CONNECTION
try {
  ob_start();
  $twitter_connection = $user->add_connection("tw","twitter",$conf["SERVICE_TOKENS"]["TWITTER"]);
  ob_end_clean();
  print_success("Add Twitter Connection");
}
catch(Exception $e){
  ob_end_clean();
  print_error("Add Twitter Connection");
  echo "\t" . $e->getMessage() . "\n";
}



// POST FACEBOOK PROFILE
$options = array(
    "receivers" => array(
        array(
            "name" => "Me"
        ),
        array(
            "name" => "Pae",
            "id" => "PAGE_ID_HERE"
        )
    ),
    "message" => array(
        "subject" => "test",
        "body" => "ABC",
        "image" => "http://politibits.blogs.tuscaloosanews.com/files/2010/07/sanford_big_dummy_navy_shirt.jpg",
        "link" => array(
            "uri" => "http://google.com",
            "description" => "link desc",
            "title" => "link title"
        )
    )
);

//POST TWEET
if($twitter_connection){
  $uris = $twitter_connection->send_message($options);
  if(count($uris) > 0){
    print_success("Post Tweet");
  }
  else {
    print_error("Post Tweet");
  }
}


//POST FB
if($fb_connection){
  $uris = $fb_connection->send_message($options);
  if(count($uris) > 0){
    print_success("Post Facebook Profile");
  }
  else {
    print_error("Post Facebook Profile");
  }
}


//TODO: POST FACEBOOK PAGE



//DELETE FACEBOOK CONNECTION
if($user->remove_connection($fb_connection->name)){
  print_success("Remove Facebook Connection");
}
else {
  print_error("Remove Facebook Connection");

}

//DELETE TWITTER CONNECTION
if($user->remove_connection($twitter_connection->name)){
  print_success("Remove Twitter Connection");
}
else {
  print_error("Remove Twitter Connection");

}

if($app->delete_user($user)){
  print_success("Delete User");
}
else {
  print_error("Delete User");
}

File: java/src/RunTest.java
import com.unificationengine.exceptions.UnificationEngineException;
import com.unificationengine.lib.message.Message;
import com.unificationengine.lib.message.MessageLink;
import com.unificationengine.lib.message.MessageOptions;
import com.unificationengine.lib.message.ReceiverTypes;
import com.unificationengine.models.UEApp;
import com.unificationengine.models.UEConnection;
import com.unificationengine.models.UEUser;

import java.io.IOException;

public class RunTest {

    public static MessageOptions getTestMessage() {
        //Specify Message Options
        MessageOptions options = new MessageOptions();

        //Add a message receiver (My wall)
        options.addReceiver(ReceiverTypes.ME);

        //Add a message receiver (My Page)
        options.addReceiver(ReceiverTypes.PAGE, "PAGE_ID");

        Message m = new Message();

        //Message Subject
        m.setSubject("Subject");

        //Message Image
        m.setImage("https://raw.githubusercontent.com/plu/JPSimulatorHacks/master/Data/test.png");

        //Message Body
        m.setBody("Java SDK Test Message!");


        //Message Link
        MessageLink mLink = new MessageLink();
        mLink.setDesc("Java SDK message desc");
        mLink.setTitle("Java SDK Msg Title");
        mLink.setUri("http://google.com");

        m.setLink(mLink);

        options.setMessage(m);
        return options;
    }


    public static void main(String[] args) throws IOException, InterruptedException {
        System.out.println(System.getProperty("user.dir"));
        Config conf = CommonWrappers.getConfig();
        UEApp app = new UEApp(conf.APP_KEY, conf.APP_SECRET);


        UEUser user = null;
        UEConnection fbConnection = null, twitterConnection = null;
        try {
            user = app.createUser();
            CommonWrappers.printSuccess("Create User");

        } catch (Exception e) {
            CommonWrappers.printError("Create User");
            //No user, we can't run the rest of the test.
            System.exit(0);
        }


        // ADD FACEBOOK CONNECTION
        try {
            fbConnection = user.addConnection("javafb","facebook", conf.FACEBOOK_TOKEN);
            CommonWrappers.printSuccess("Add Facebook Connection");
        } catch (Exception e) {
            CommonWrappers.printError("Add Facebook Connection");
            System.out.println("\t" + e.getMessage());
        }


        // ADD TWITTER CONNECTION
        try {
            twitterConnection = user.addConnection("javatwitter","twitter", conf.TWITTER_TOKEN);
            CommonWrappers.printSuccess("Add Twitter Connection");
        } catch (Exception e) {
            CommonWrappers.printError("Add Twitter Connection");
            System.out.println("\t" + e.getMessage());
        }



        // Post Facebook Profile
        if(fbConnection != null) {
            try {
                fbConnection.sendMessage(getTestMessage());
                CommonWrappers.printSuccess("Post Facebook Profile");
            } catch (Exception e) {
                CommonWrappers.printError("Post Facebook Profile\"");
                System.out.println("\t" + e.getMessage());
            }
        }
        // Post Facebook Page
        // Post Tweet
        if(twitterConnection != null) {
            try {
                twitterConnection.sendMessage(getTestMessage());
                CommonWrappers.printSuccess("Post Tweet");
            } catch (Exception e) {
                CommonWrappers.printError("Post Tweet");
                System.out.println("\t" + e.getMessage());
            }
        }




        // Delete Facebook Connection
        if(fbConnection != null ) {
            try {
                user.removeConnection(fbConnection);
                CommonWrappers.printSuccess("Remove Facebook Connection");
            } catch (UnificationEngineException e) {
                e.printStackTrace();
                CommonWrappers.printError("Remove Facebook Connection");
                System.out.println("\t" + e.getMessage());
            }
        }
        // Delete Twitter Connection
        if(twitterConnection != null) {
            try {
                user.removeConnection(twitterConnection);
                CommonWrappers.printSuccess("Remove Twitter Connection");
            } catch (UnificationEngineException e) {
                e.printStackTrace();
                CommonWrappers.printError("Remove Twitter Connection");
                System.out.println("\t" + e.getMessage());
            }

        }
        // Delete User
        try {
            app.deleteUser(user.getUri());
            CommonWrappers.printSuccess("Delete User");
        } catch (UnificationEngineException e) {
            CommonWrappers.printError("Delete User");
            System.out.println("\t" + e.getMessage());
        }

    }


}
